### PR TITLE
Bind repeated SQL values once, and show stored payment anomalies on the system map

### PR DESCRIPTION
**Pull request by stefan-burke**
## What changed

**One way to bind a value that a statement uses more than once.** A new helper, `numberedStatement` (`src/shared/db/numbered-statement.ts`), builds a statement through a `bind` function. Each bound value gets a numbered slot (`?1`, `?2`), and the returned token can appear in the SQL text many times. A branded token type (`SqlParameterToken`) marks the seams where only a bound value may enter the SQL. Around twenty database modules now build their repeated-bind statements this way: capacity checks, attendee create and update, groups, images, the migration lock, payment anchors, refund authority, pruning, and the payment snapshot. The helper also owns the shared `allConditions` AND-join and the `ALWAYS_TRUE` guard, so each is written once.

**The capacity SQL is rebuilt on the helper.** `src/shared/db/capacity.ts` derives the listing and group count subqueries and the whole-cart batch check from bound tokens. One shared `demandClauses` walk now serves both the listing and the group demand maps.

**Stored payment and SumUp anomalies show on the system map.** A typed scanner (`src/shared/db/schema-anomaly-scan.ts`) finds records whose stored words the payment machines do not know — for example a SumUp checkout whose `recovery_state` the recovery queue can never select — and `/admin/schema` lists each hit with its record id. This completes the "Surface unknown SumUp recovery states" TODO entry.

**Review hardening.** The capacity insert takes its attendee id expression as a `NumberedSql`, so the expression and its value cannot drift apart. `existsSql` takes the branded token, so a raw string cannot reach that seam. The copied group price statement binds its price-type words once. Test helpers now check that a row exists instead of a non-null assertion (`requireAttendee`, `requireOne`).

**Housekeeping.** The invite prune scan binds its checkpoint once as `?1`. The stale "14 survivors in `listing-parents.ts`" TODO entry is corrected and closed: the full mirror suite already killed twelve, and the two real survivors are recorded as equivalent with proofs. The numbered-parameter sweep results are written into `TODO.md` as a site list for later conversions. New equivalent-mutant records cover the sweep's unkillable operators, each with its reason.

## Checks

- `deno task precommit` (typecheck, lint, duplication, copy checks, full suite) passed.
- Targeted mutation runs: `src/shared/db/listing-parents.ts` scored 43/43 and `src/shared/db/prune.ts` scored 66/66, with recorded equivalents suppressed.
- `deno task precommit:mutation` over every changed source is in progress; its result will be reported on this pull request.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_013Ve7CEwp4N5YhvfgumG7KN

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved database operations involving repeated values, capacity checks, attendee updates, payments, groups, and image links.
  * Image links now consistently verify that both the image and destination exist.
  * Payment snapshots now avoid using attendees associated with unrelated ledger events.
  * Improved validation for attendee bookings, group memberships, payment references, and capacity limits.

* **Maintenance**
  * Removed obsolete accounting and database query utilities.
  * Expanded automated coverage for booking, payment, capacity, image, group, and accounting scenarios.

### CODE REVIEW

**Comment by stefan-burke** [comment]
@coderabbitai review

The earlier automatic review was rate limited. The limit has reset, so this comment requests the review.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/shared/db/groups.ts [hunk 1]
```diff
@@ -732,27 +734,24 @@ export const copyPackageMemberOverridesTx = async (
...
+             AND sourcePrice.price_type IN (${bind(PRICE_TYPE_GROUP)}, ${bind(PRICE_TYPE_GROUP_DAY)})
+              AND EXISTS (
+                SELECT 1 FROM group_listings AS groupListing
+                 WHERE groupListing.listing_id = ${newId}
+                   AND ((sourcePrice.price_type = '${PRICE_TYPE_GROUP}'
+                           AND sourcePrice.price_id = CAST(groupListing.group_id AS TEXT))
+                     OR (sourcePrice.price_type = '${PRICE_TYPE_GROUP_DAY}'
+                           AND sourcePrice.price_id LIKE (groupListing.group_id || '/%')))
+              )`;
+    }),
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/db/groups.ts:755]: Fixed in dac5e7f. Each price-type word is now bound once and its token is reused in the IN clause and both EXISTS arms. The duplicate-listing tests cover the statement (`test/features/admin/listings-edit/duplicate.test.ts`).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/create/result.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  createDailyTestListing,
+  createTestListing,
+} from "#test-utils/db-helpers/listings.ts";
+
+const requireAttendee = (
+  result: Awaited<ReturnType<typeof createAttendeeAtomic>>,
+) => {
+  if (!result.success) throw new Error("expected attendee creation to succeed");
+  return result.attendees[0]!;
+};
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/create/result.test.ts]: Fixed in dac5e7f. A shared `requireAttendee` helper now lives in `#test-utils/db-helpers/attendee-creation.ts`: it asserts the create succeeded and that the first attendee is really there, with a clear message for each. Both test files use it, and the non-null assertions are gone.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/test-utils/db-helpers/attendee-creation.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  await expectNothingWritten(listingId, transferCount);
+};
+
+export const storedEventGroup = async (attendeeId: number): Promise<string> => {
+  const row = await queryOne<{ ledger_event_group: string }>(
+    "SELECT ledger_event_group FROM listing_attendees AS attendee WHERE attendee_id = ?",
+    [attendeeId],
+  );
+  return row!.ledger_event_group;
+};
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/test-utils/db-helpers/attendee-creation.ts:155]: Fixed in dac5e7f. `storedEventGroup` now reads with `requireOne`, so a missing row reports the failed query instead of a property access on null.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/attendees/capacity/checks/insert.test.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      {
+        date: "2026-06-24",
+        durationDays: 2,
+        listingId: 17,
+        orderToken: "order-17",
+        packageGroupId: 23,
+        parentListingId: 19,
+        quantity: 3,
+      },
+      (bind) => `attendee_id = ${bind(41)}`,
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/attendees/capacity/checks/insert.test.ts]: Fixed in f59c16f. The callback now returns `bind(41)` directly, so the statement selects the attendee id itself. The same commit answers the outside-diff comment from this review: the `WHERE`-omission assertion moved to its own overbooking test, so each test has one reason to fail.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by stefan-burke** [comment]
@coderabbitai review

The previous automatic review stopped because the head moved during it. The head is now stable at f67f4e0, so this comment requests the review.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/shared/db/groups/set-groups.test.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+    // A mixed submission applies the member entry and drops the non-member.
+    await setGroupPackageMembers(group.id, [
+      { listingId: member.id, price: 700 },
+      { listingId: outsider.id, price: 999 },
+    ]);
+    expect(await getTestPackagePrices(group.id)).toEqual(
+      new Map([[member.id, 700]]),
+    );
+  });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/shared/db/groups/set-groups.test.ts]: Fixed in 0f20512. The combined case is now two tests over one shared fixture: an outsider-only submission stays a no-op, and a mixed submission applies the member entry and drops the non-member.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/test-utils/db-helpers/attendee-creation.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  assert(result.success, "Expected attendee creation to succeed");
+  return result;
+};
+
+/** The one attendee a successful create returned, checked rather than assumed. */
+export const requireAttendee = (result: CreateAttendeeResult): Attendee => {
+  assert(result.success, "Expected attendee creation to succeed");
+  const attendee = result.attendees[0];
+  assert(attendee !== undefined, "Expected the create to return one attendee");
+  return attendee;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/test-utils/db-helpers/attendee-creation.ts:46]: Fixed in 0f20512. `requireAttendee` now asserts the create returned exactly one attendee before it hands the attendee back.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by stefan-burke** [comment]
The promised `deno task precommit:mutation` result over every changed source: **99.9% — 1187 of 1188 mutants detected** (1245 total, 57 suppressed as recorded equivalents).

The one survivor is the `hasChildren` ternary in `validateListingGroupMembershipTx`: this PR's test split moved its only killing test into a file that `groups.ts` owns, so `membership.ts`'s own suite lost it. #2126 closes it with two direct cases in `membership`'s suite.

The same local run also flagged `storedEventGroup` as uncovered, but that was a race with edits made during the run — CI's coverage check passed on the merged head, and the helper's callers are in place.

---
_Generated by [Claude Code](https://claude.ai/code)_
